### PR TITLE
Update login logic using redux

### PR DESCRIPTION
**Pull request by powermukisa**
-We're now able to handle authentication in a much simpler way, thanks to redux
-Also added logic to only allow Tala emails to access gezako, and a set of test emails(non tala emails)
-Added a company collection so that all Tala data is stored under the Tala document in firestore

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -15,54 +14,43 @@ import CreateSpockReport from "./components/reports/create/CreateSpockReport";
...
-                              </React.Fragment>
-
-                              : <Route path='/' exact component={Login}/>
-                          }
-                      </React.Fragment>
-                  : <Route path='/' exact component={null}/>
-                }
-              </Switch>
-            </BrowserRouter>
+              <Route path='/login' component={Login}/>
```
Comment (powermukisa) [inline src/App.js]: we now have a dedicated login page url

File: src/components/home/Home.js [hunk 2]
```diff
@@ -85,7 +86,10 @@ class Home extends Component {
     }
 
     render() {
-        return (
+      const { auth } = this.props;
+      if (!auth.uid) return <Redirect to='/login' />
```
Comment (powermukisa) [inline src/components/home/Home.js]: show login page if user is not logged in

File: src/index.js [hunk 3]
```diff
@@ -18,14 +18,17 @@ const store = createStore(
...
     compose(
         applyMiddleware(thunk.withExtraArgument({getFirebase, getFirestore})),
-        reactReduxFirebase(fbConfig),
+        reactReduxFirebase(fbConfig, {attachAuthIsReady: true}),
         reduxFirestore(fbConfig)
     )
 );
 
+
+store.firebaseAuthIsReady.then(() => {
```
Comment (powermukisa) [inline src/index.js]: this load the DOM only when firebase auth is ready

File: src/App.js (hunk 1 above)
Comment (derekleiro) [reply src/App.js]: Will it start from here the redirect to "/"?

Comment (powermukisa) [reply src/App.js]: yes it will. once `/` is loaded it will check if user is logged in. if not, we'll redirect to `/login`

**Comment by powermukisa** [comment]
Merging without approval (should be approved by @derekleiro )

**Pull request merged**